src/test/java/com/github/tomakehurst/wiremock/common/FilenameMakerTest.java

```diff
@@ -1,66 +1,79 @@
+/*
+ * Copyright (C) 2023 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.github.tomakehurst.wiremock.common;
 
-import com.github.tomakehurst.wiremock.client.WireMock;
-import com.github.tomakehurst.wiremock.common.filemaker.FilenameMaker;
-import com.github.tomakehurst.wiremock.stubbing.StubMapping;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-
 import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
 import static com.github.tomakehurst.wiremock.client.WireMock.ok;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.common.filemaker.FilenameMaker;
+import com.github.tomakehurst.wiremock.stubbing.StubMapping;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
 public class FilenameMakerTest {
 
-    private FilenameMaker filenameMaker;
-    @BeforeEach
-    public void init() {
-        filenameMaker = new FilenameMaker();
-    }
+  private FilenameMaker filenameMaker;
+
+  @BeforeEach
+  public void init() {
+    filenameMaker = new FilenameMaker();
+  }
 
-    @Test
-    public void generatesNameFromStubNameWhenPresent() {
-        StubMapping mapping =
-                WireMock.get("/named").withName("This is a NAMED stub").willReturn(ok()).build();
+  @Test
+  public void generatesNameFromStubNameWhenPresent() {
+    StubMapping mapping =
+        WireMock.get("/named").withName("This is a NAMED stub").willReturn(ok()).build();
 
-        assertThat(
-                filenameMaker.filenameFor(mapping),
-                is("get-named-" + mapping.getId() + ".json"));
-    }
+    assertThat(filenameMaker.filenameFor(mapping), is("get-named-" + mapping.getId() + ".json"));
```

Comment:
We seem to have lost the logic for including the name when it's present - this only uses the method + URL + ID.

Reply:
Added logic for smart detection `name` inside the stub and decoration existing template. Also added logic for detection `url` or `urlPath` because it could be both cases.